core/src/OpenMPTarget/Kokkos_OpenMPTarget_ParallelFor_Team.hpp

```diff
@@ -140,8 +140,16 @@ class ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties...>,
 // guarantees that the number of teams specified in the `num_teams` clause is
 // always less than or equal to the maximum concurrently running teams.
 #if !defined(KOKKOS_IMPL_OPENMPTARGET_HIERARCHICAL_INTEL_GPU)
+      // Use scratch memory extensions to request dynamic shared memory for the
+      // right compiler/architecture combination.
+#if defined(KOKKOS_IMPL_OPENMPTARGET_LLVM_EXTENSIONS)
+#pragma omp target teams thread_limit(team_size) firstprivate(a_functor) \
+    num_teams(max_active_teams) is_device_ptr(scratch_ptr)               \
+        ompx_dyn_cgroup_mem(shmem_size_L0)
+#else
```

Comment:
Considering all the use cases are just "add the directive if LLVM extensions are available", you could do something along these lines (not tested)
```C++
#define KOKKOS_IMPL_OMPTARGET_PRAGMA_HELPER(x) _Pragma(#x)
#define KOKKOS_IMPL_OMPTARGET_PRAGMA(x) KOKKOS_IMPL_OMP_PRAGMA_HELPER(omp target x)

#ifdef KOKKOS_IMPL_OPENMPTARGET_LLVM_EXTENSIONS
#define KOKKOS_IMPL_OMPX_DYN_CGROUP_MEM(N) ompx_dyn_cgroup_mem(N)
#else
#define KOKKOS_IMPL_OMPX_DYN_CGROUP_MEM(N)
#endif

KOKKOS_IMPL_OMPTARGET_PRAGMA(teams thread_limit(team_size) firstprivate(a_functor) \
    num_teams(max_active_teams) is_device_ptr(scratch_ptr) \
    KOKKOS_IMPL_OMPX_DYN_CGROUP_MEM(shmem_size_L0))
#pragma omp parallel
    {
       // ...

Reply:
For now, I wrapped only the pragmas in this file. In a future PR I will put the macro in a central file and update this in all openmp calls.